utils/machine/machine.go

```diff
@@ -0,0 +1,259 @@
+var CpuClockSpeedMHz = regexp.MustCompile("cpu MHz\\t*: +([0-9]+.[0-9]+)")
+var memoryCapacityRegexp = regexp.MustCompile("MemTotal: *([0-9]+) kB")
+var swapCapacityRegexp = regexp.MustCompile("SwapTotal: *([0-9]+) kB")
+
+// GetClockSpeed returns the CPU clock speed, given a []byte formatted as the /proc/cpuinfo file.
+func GetClockSpeed(procInfo []byte) (uint64, error) {
+	// First look through sys to find a max supported cpu frequency.
+	const maxFreqFile = "/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq"
+	if utils.FileExists(maxFreqFile) {
+		val, err := ioutil.ReadFile(maxFreqFile)
+		if err != nil {
+			return 0, err
+		}
+		var maxFreq uint64
+		n, err := fmt.Sscanf(string(val), "%d", &maxFreq)
+		if err != nil || n != 1 {
+			return 0, fmt.Errorf("could not parse frequency %q", val)
+		}
+		return maxFreq, nil
+	}
+	// Fall back to /proc/cpuinfo
+	matches := CpuClockSpeedMHz.FindSubmatch(procInfo)
+	if len(matches) != 2 {
+		//Check if we are running on Power systems which have a different format
+		CpuClockSpeedMHz, _ = regexp.Compile("clock\\t*: +([0-9]+.[0-9]+)MHz")
+		matches = CpuClockSpeedMHz.FindSubmatch(procInfo)
+		if len(matches) != 2 {
+			return 0, fmt.Errorf("could not detect clock speed from output: %q", string(procInfo))
+		}
+	}
+	speed, err := strconv.ParseFloat(string(matches[1]), 64)
+	if err != nil {
+		return 0, err
+	}
+	// Convert to kHz
+	return uint64(speed * 1000), nil
+}
+
+// GetMachineMemoryCapacity returns the machine's total memory from /proc/meminfo.
+// Returns the total memory capacity as an int64 (number of bytes).
+func GetMachineMemoryCapacity() (int64, error) {
+	out, err := ioutil.ReadFile("/proc/meminfo")
+	if err != nil {
+		return 0, err
+	}
+
+	memoryCapacity, err := getMemoryCapacity(out)
+	if err != nil {
+		return 0, err
+	}
+	return memoryCapacity, err
+}
+
+// getSwapCapacity applies the swapCapacity regexp to its input []byte.
+// Returns the total memory capacity as an int64 (number of bytes).
+func getMemoryCapacity(b []byte) (int64, error) {
+	matches := memoryCapacityRegexp.FindSubmatch(b)
+	if len(matches) != 2 {
+		return -1, fmt.Errorf("failed to find memory capacity in output: %q", string(b))
+	}
+	m, err := strconv.ParseInt(string(matches[1]), 10, 64)
+	if err != nil {
+		return -1, err
+	}
+
+	// Convert to bytes.
+	return m * 1024, err
+}
+
+// GetMachineSwapCapacity returns the machine's total swap from /proc/meminfo.
+// Returns the total swap capacity as an int64 (number of bytes).
+func GetMachineSwapCapacity() (int64, error) {
+	out, err := ioutil.ReadFile("/proc/meminfo")
+	if err != nil {
+		return 0, err
+	}
+
+	swapCapacity, err := getSwapCapacity(out)
+	if err != nil {
+		return 0, err
+	}
+	return swapCapacity, err
+}
+
+// getSwapCapacity applies the swapCapacity regexp to its input []byte.
+// Returns the total swap capacity as an int64 (number of bytes).
+func getSwapCapacity(b []byte) (int64, error) {
```

Comment:
why not merge getSwapCapacity and getMemoryCapacity into parseMemory and move error reporting to the public methods? It will avoid some duplicated code.